src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -951,10 +951,18 @@ public static IEdmEntitySet FindDeclaredEntitySet(this IEdmModel model, string q
             if (!model.TryFindContainerQualifiedEntitySet(qualifiedName, out foundEntitySet))
             {
                 // try searching by entity set name in container and extended containers:
-                IEdmEntityContainer container = model.EntityContainer;
-                if (container != null)
+                try
+                {
+                    IEdmEntityContainer container = model.EntityContainer;
+                    if (container != null)
+                    {
+                        return container.FindEntitySetExtended(qualifiedName);
+                    }
+                }
+                catch (NotImplementedException)
                 {
-                    return container.FindEntitySetExtended(qualifiedName);
+                    // model.EntityContainer can throw NotImplementedException
+                    return foundEntitySet;

```

Comment:
>foundEntitySet [](start = 27, length = 14)

Is this null in this case? Would it make more sense to be explicit and return null?  Same below.